sdk/core/azure-core/azure/core/pipeline/transport/base.py

```diff
@@ -81,6 +84,37 @@ def _urljoin(base_url, stub_url):
     return parsed.geturl()
 
 
+class _HTTPSerializer(HTTPConnection):
+    """Hacking the stdlib HTTPConnection to serialize HTTP request as strings.
+    """
+    def __init__(self, *args, **kwargs):
+        self.skip_host = kwargs.pop("skip_host", True)
+        self.skip_accept_encoding = kwargs.pop("skip_accept_encoding", True)
+        self.buffer = b''
+        kwargs.setdefault("host", "fakehost")
```

Comment:
Is there a reason why provide a dummy host name?

Reply:
Yes, I use this to generate a body, and I don't care the headers this guy will generate.